server/src/main/java/io/druid/metadata/SQLMetadataRuleManager.java

```diff
@@ -287,9 +292,19 @@ public void poll()
       log.info("Polled and found rules for %,d datasource(s)", newRules.size());
 
       rules.set(newRules);
+      retryStartTime = 0;
     }
     catch (Exception e) {
-      log.error(e, "Exception while polling for rules");
+      if (retryStartTime == 0) {
+        retryStartTime = System.currentTimeMillis();
+      }
+
+      if (new DateTime(retryStartTime).plus(config.get().getAlertThreshold()).isBefore(new DateTime())) {
```

Comment:
instead of wrapping and unwrapping, can we do System.currentTimeMillis() - retryStartTime > alertThreshold?